### PR TITLE
Integration spec

### PR DESCRIPTION
**Pull request by AbdollahRaafat**
- Write integration tests with Capybara gem.
- Secure app from n+1 problems.

### CODE REVIEW

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    it 'I can see how many comments a post has.' do
+      expect(page).to have_content(@post1.comments_counter)
+    end
+
+    it 'I can see how many likes a post has.' do
+      expect(page).to have_content(@post1.likes_counter)
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/features/post_index_spec.rb]: - [ ] [REQUIRED] Hello team, some of the tests are missing for this file, kindly add
* Tests that assert you can see a post's title.
* Tests that assert you can see some of the post's body.
* Tests that assert you can see a section for pagination if there are more posts than fit on the view.
* Tests that redirect to the post's show page when you click on a post.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi Team,
@AbdollahRaafat and @Ney2 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- All tests are passing ✔️
- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb (hunk 1 above)
Comment (Ney2) [reply spec/features/post_index_spec.rb]: Thank you for your insightful feedback and for reviewing my project

**Comment by Ney2** [comment]
Thank you for approving my project

**Pull request merged**